### ISSUE TITLE
Custom SSL certificates for HTTPS

### ISSUE THREAD

**Issue by nadavgolden** [appwrite/appwrite#1084]
## 🚀 Feature
Enable having custom certificates from a local CA server instead of Certbot.

## Have you spent some time to check if this issue has been raised before?
Searched documentation and issues history here, didn't find any similar issues.
Read some of the source code and available config flags, didn't see any that are relevant.
Went to the source code and it seems like Certbot is the only option.

### Have you read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/master/CODE_OF_CONDUCT.md)?
Yes.

## Pitch
In my specific case I have an offline environment with no access to Let's Encrypt's servers.
Also, we use a custom CA for our servers so I'd rather use our own than using Let's Encrypt.

Please enlighten me if there is a way to configure Certbot (or AppWrite) to use custom certificates.

**Comment by nadavgolden** [comment]
I guess you can use a reverse proxy with a custom certificate so that:
`AppWrite (self-signed)<- nginx (trust insecure upstream connection; use custom certificate) <- client (HTTPS)`
But I'd like to avoid it

**Comment by eldadfux** [comment]
> I guess you can use a reverse proxy with a custom certificate so that:
> `AppWrite (self-signed)<- nginx (trust insecure upstream connection; use custom certificate) <- client (HTTPS)`
> But I'd like to avoid it

Agree, this can be a great solution. You can also use the Traefik container to set custom certificates like explained here:
https://medium.com/@clintcolding/use-your-own-certificates-with-traefik-a31d785a6441

**Comment by eldadfux** [comment]
We would love help with this issue. All required is to add env vars that will allow our Swoole HTTP server to pick SSL certs from a specific location without having to rely on Traefik for SSL.

**Comment by AmrDhaliwal** [comment]
Hi, I was hoping to work on this issue?

**Comment by HariniKrishnan** [comment]
hi can I work on this issue?!

**Comment by davedawkins** [comment]
Would the resolution of this issue allow me to use certificates I already have installed on my server? I've installed AppWrite on `abc.dev`, and assigned it ports 8585 (http) & 8543 (https). (On this server, AppWrite cannot have port 80 unless I route "/v1/..."). Because it's ".dev", the self-signed certificate isn't accepted by the browser.  I was wondering if I could copy the certificates used by the main website into the AppWrite server.

**Comment by christyjacob4** [comment]
@bodnar1212 its not implemented currently, but as @eldadfux  mentioned, we can introduce a new env variable that points to the location of the SSL certificates instead of replying on the default location.

**Comment by SingHar** [comment]
Hi,

It would be great if we can use our own SSL certificates.

A decent use case is when using Cloudflare provided SSL certificates.

**Comment by binaryfire** [comment]
Hi all. So if I understand correctly, there's no way to use our own SSL certs? I'm surprised there's no option to disable the Lets Encrypt integration. Reverse proxying through Cloudflare or AWS Cloudfront is standard practice nowadays, especially for larger organizations.
